providers/google.go

```diff
@@ -265,6 +268,35 @@ func getAdminService(opts options.GoogleOptions) *admin.Service {
 	return adminService
 }
 
+func getTargetPrincipal(ctx context.Context, opts options.GoogleOptions) (targetPrincipal string) {
+	targetPrincipal = opts.TargetPrincipal
+
+	if targetPrincipal == "" {
```

Comment:
Nit, if you reverse this it will make the flow easier to follow and you can reduec the indent
```suggestion
	if targetPrincipal != "" {
		return targetPrincipal
	}
```

Reply:
Great suggestion! Thx!